src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -1310,6 +1326,10 @@ private static bool HadLambdaConversionError(BindingDiagnosticBag diagnostics, A
                     }
                 }
             }
+
+            static bool isStringLiteralToInterpolatedStringHandlerArgumentConversion(BoundExpression argument, ParameterSymbol parameter)
+                => argument.Syntax.Kind() is SyntaxKind.StringLiteralExpression &&
```

Comment:
```suggestion
                => argument is BoundLiteral { Type.SpecialType: SpecialType.String } &&
```

We should also consider whether we want to try and handle the case where you do `"a" + "b"`, and treat that as if it was a string (or `"a" + $"b"`). This would require a walk of the tree to determine that though, and it may be too complicated for a "make this error nicer" scenario. Up to you.

Reply:
I don't think such niche scenario is worth code change